lib/ecto/adapters/sql.ex

```diff
@@ -317,10 +317,9 @@ defmodule Ecto.Adapters.SQL do
     |> Keyword.put_new(:pool_timeout, @pool_timeout)
   end
 
-  defp normalize_pool(Ecto.Adapters.SQL.Sandbox),
-    do: DBConnection.Ownership
-  defp normalize_pool(pool),
-    do: pool
+  defp normalize_pool(pool) do
+    if function_exported?(pool, :unboxed_run, 2), do: DBConnection.Ownership, else: pool
```

Comment:
I don't think we should do this change. We only do it for backwards compatibility, your sandbox should require DBConnection.Ownership to be explicitly passed instead.